### ISSUE TITLE
[BUG] Reloading the buffer appends the virtual text

### ISSUE THREAD

**Issue by alexaandru** [vuki656/package-info.nvim#37]
### Issues

- [X] I have checked [existing issues](https://github.com/vuki656/package-info.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Plugin Version

- [X] I am using the latest version of the plugin

### Neovim Version

- [X] I am using the 0.5 neovim version

### Neovim Version

NVIM v0.6.0-dev+112-ge7bde44d6

### Branch

master

### Actual behavior

With an opened `package.json` hit `:e`. With each hit, text gets appended, and ends up looking like this:

![Screenshot from 2021-08-09 11-37-41](https://user-images.githubusercontent.com/85237/128679567-4105cf86-c943-458d-8a93-2d907f5c644b.png)

### Expected behavior

I would expect virtual text to be replaced on reloading the buffer.

### Steps to reproduce

See "Actual behavior"

### Help

No, sorry.

**Comment by vuki656** [comment]
Hey, thanks for reporting.

Although I couldn't reproduce on my end, I pushed a commit that should clear any package-info virtual text before setting a new one. 

Let me know if you are still experiencing issues.

**Comment by vuki656** [comment]
Hmm, it seems that clearing virtual text before setting a new one completely breaks the plugin/prevents new text from being set. I reverted the "fix'.

I'll have to give it another shot. In the meanwhile, could you post your `require("package-info").setup()` code so I can try and replicate it with the same config?

**Comment by alexaandru** [comment]
Sure, here it is: https://github.com/alexaandru/nvim-config/blob/wip/fnl/config/package-info.fnl and here is translated back to `Lua`: 

```Lua
{
  autostart = true,
  colors = {
    outdated = "#f1aa66",
    up_to_date = "#4C5058"
  },
  icons = {
    enable = true,
    style = {
      outdated = " 🚩",
      up_to_date = " "
    }
  }
}
```

as for actually calling setup, that's just `require("package-info").setup(require("config.package-info"))`

I just retried with no settings at all (just `setup()`) and it does the same thing, no change.

**Comment by vuki656** [comment]
Does fennel transpile it at runtime or? 

If you have a Lua file, could you try putting the config there manually just to test if it would work that way?

I'm guessing it has something to do with how and when transpiring to Lua is happening.

**Comment by alexaandru** [comment]
Yes it does, it all works like a charm, all my config is now 100% Fennel, thanks to [Hotpot](https://github.com/rktjmp/hotpot.nvim).

Anyway, sure, to get it out of the equation, I can make a Lua config file, let's see... there you go, I ran `nvim -u ~/cfg.lua package.json` where `cfg.lua` contains:

```Lua
vim.cmd("packadd package-info")
require("package-info").setup()
```

So that would be 100% Lua, Fennel completely out of the equation (and also the rest of my own config). It behaves exactly the same, keeps adding the virtual text.

**Comment by vuki656** [comment]
Okay, thanks.

Ill let you know once it's fixed.

**Comment by alexaandru** [comment]
No worries, it's nothing critical anyway, the plugin is great! :) I can take a look at it, see if I can come up with a fix, but only in the weekend. The only thing I can think of, is keeping track of extmark IDs and then passing that to `vim.api.nvim_buf_set_extmark()`, as in their absence, it will always create new marks. That's the only way to update them. OR... keep some buffer local bool, to signal that the marks were already loaded there, and then don't create them if that bool is set.

**Comment by vuki656** [comment]
It is currently using the same namespace id that's why it's strange. 

I'll have to play around with it.

**Comment by alexaandru** [comment]
It's not the namespace id I was referring to, but the extmark id, that is different. That is what you can pass in the last, `opts` parameter of [`nvim_buf_set_extmark()`](https://neovim.io/doc/user/api.html#nvim_buf_set_extmark()).

**Comment by vuki656** [comment]
Hey, can you try the `develop` branch and see if it's fixed.

It should now clear everything before setting new text.

I also added a bunch of new features that you can find in the readme.

It's still kinda buggy, so ill be testing it for a week before merging to master.

**Comment by alexaandru** [comment]
Hello, 

I tested with the latest master, and there it's fixed already :) 

Then I tried the `develop` branch, but that one is crashing for me with: `E5108: Error executing lua ...im/pack/plugins/opt/package-info/lua/package-info/ui.lua:1: module 'nui.menu' not found`.

I will close this ticket, since the issue is fixed on master. Happy to help with testing that develop branch when you push the fix for the above.

Cheers!
Alex
